app/scripts/migrations/165.ts

```diff
@@ -0,0 +1,63 @@
+import { hasProperty, isObject } from '@metamask/utils';
+import { cloneDeep } from 'lodash';
+import { PreferencesControllerState } from '../controllers/preferences-controller';
+
+export type VersionedData = {
+  meta: {
+    version: number;
+  };
+  data: {
+    PreferencesController?: Partial<PreferencesControllerState> & {
```

Comment:
Nit: I would recommend inlining a minimal type here (e.g. drop `Partial<PreferencesControllerState>`). Importing state types in migrations usually causes more harm than good because we need to assume when writing the migration that state might be corrupted by the previous one, and because the state type changes over time but the migration must remain valid for this specific point in time.